src/lib.rs

```diff
@@ -607,19 +570,65 @@ pub(crate) fn calculate_fee(txns: &[TransactionNote]) -> Result<u64, TxnApiError
         .ok_or_else(|| TxnApiError::IncorrectFee("Overflow in total fee".to_string()))
 }
 
+/// Compute signature over a list of receiver memos
+pub fn sign_receiver_memos(
+    keypair: &KeyPair,
+    recv_memos: &[ReceiverMemo],
+) -> Result<Signature, TxnApiError> {
+    let digest = get_receiver_memos_digest(recv_memos)?;
+    Ok(keypair.sign(&[digest]))
+}
+
 #[cfg(test)]
 mod test {
     use crate::{
-        calculate_fee, derive_txns_fee_record,
+        calculate_fee, derive_txns_fee_records,
         errors::TxnApiError,
         keys::{UserKeyPair, UserPubKey},
-        prepare_txns_fee_record,
-        structs::{AssetDefinition, BlindFactor, FreezeFlag},
+        structs::{AssetDefinition, BlindFactor, FreezeFlag, RecordOpening},
         txn_batch_verify,
         utils::params_builder::TxnsParams,
         KeyPair, TransactionNote,
     };
-    use ark_std::{vec, vec::Vec};
+    use ark_std::{
+        rand::{CryptoRng, RngCore},
+        string::ToString,
+        vec,
+        vec::Vec,
+    };
+
+    /// Prepare the record opening corresponding to collected fee from a list of
+    /// transaction notes (within a block). The result is a record opening whose
+    /// blinding factor should be directly appended to the same block before
+    /// being disseminated to other nodes. This function is intended to be
+    /// called by nodes proposing blocks.
+    ///
+    /// * `txns` - List of verified transaction notes within a block
+    /// * `owner_pk` - Public key of the owner of the collected fee, usually the
+    ///   block proposer's public key
+    pub(crate) fn prepare_txns_fee_record<R>(
+        rng: &mut R,
+        txns: &[TransactionNote],
+        owner_pk: UserPubKey,
+    ) -> Result<RecordOpening, TxnApiError>
```

Comment:
why are we still having this?

Reply:
weird. I removed it here 
https://github.com/EspressoSystems/cap/pull/34/commits/90bc8a73bb8f76be56efd2edc838d1d6f8ccc70b

Reply:
wait, it is used by testing only.